libraries/chain/controller.cpp

```diff
@@ -278,6 +291,22 @@ struct block_data_t {
       }, v);
    }
 
+   signed_block_ptr fork_db_fetch_block_by_num(uint32_t block_num) const {
+      return std::visit([&](const auto& bd) -> signed_block_ptr {
+         auto bsp = bd.fork_db.search_on_branch(bd.fork_db.head()->id(), block_num);
+         if (bsp) return bsp->block;
+         return {};
+      }, v);
+   }
+
+   std::optional<block_id_type> fork_db_fetch_block_id(uint32_t block_num) const {
```

Comment:
Can be named as `fork_db_fetch_block_id_by_num` to match `fork_db_fetch_block_by_num`.